File: 2024/src/03/day03a.py
import re

pattern = re.compile(r"mul\((\d+),(\d+)\)")

result = 0
with open("input/day03.in", "r") as file:
    for line in file:
        for match in pattern.finditer(line):
            result += int(match[1]) * int(match[2])
print(result)

File: 2024/src/03/day03a.js
const fs = require("fs");

fs.readFile("input/day03.in", "utf8", (err, data) => {
  let result = 0;
  const pattern = /mul\((\d+),(\d+)\)/g;
  const lines = data.split("\n");
  lines.forEach((line) => {
    const matches = line.matchAll(pattern);
    for (const match of matches) {
      result += Number(match[1]) * Number(match[2]);
    }
  });
  console.log(result);
});
